### PR TITLE
Update witnesses for transaction signature 

### PR DESCRIPTION
**Pull request by duanyytop**
- Add witness data structure
- Add single key single signature tx example
- Add multi key single signature tx example

### CODE REVIEW

File: serialization/src/main/java/org/nervos/ckb/type/fixed/Option.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package org.nervos.ckb.type.fixed;
+
+import org.nervos.ckb.type.base.FixedType;
+import org.nervos.ckb.type.base.Type;
+
+/** Copyright © 2019 Nervos Foundation. All rights reserved. */
+public class Option extends FixedType<Type> {
```
Comment (shaojunda) [inline serialization/src/main/java/org/nervos/ckb/type/fixed/Option.java]: I recommend using a more specific name, like `BytesOpt`, `option` in serialization RFC is dynamic size, and is a `Composite Types`. https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0008-serialization/0008-serialization.md#option

Comment (duanyytop) [reply serialization/src/main/java/org/nervos/ckb/type/fixed/Option.java]: `BytesOpt` can be implemented by`new Option(new Bytes(value))`, and `Option` here can be dynamic.

You can see https://github.com/nervosnetwork/ckb-sdk-java/blob/update-witnesses/core/src/main/java/org/nervos/ckb/utils/Serializer.java#L94-L100

File: core/src/main/java/org/nervos/ckb/transaction/CellsWithLock.java [hunk 2]
```diff
@@ -1,14 +1,14 @@
-package org.nervos.ckb.example.transaction;
+package org.nervos.ckb.transaction;
 
 import java.util.List;
 import org.nervos.ckb.type.cell.CellInput;
 
 /** Copyright © 2019 Nervos Foundation. All rights reserved. */
-public class CellsWithPrivateKey {
+public class CellsWithLock {
```
[the remark this answers is not in the document]
Comment (duanyytop) [reply core/src/main/java/org/nervos/ckb/transaction/CellsWithLock.java]: Lock(lock script) is more general and if we put more information about account, I think private key is not appropriate.

[the remark this answers is not in the document]
Comment (shaojunda) [reply core/src/main/java/org/nervos/ckb/transaction/CellsWithLock.java]: but now only privateKey?

File: serialization/src/main/java/org/nervos/ckb/type/fixed/Option.java (hunk 1 above)
Comment (shaojunda) [reply serialization/src/main/java/org/nervos/ckb/type/fixed/Option.java]: why put option under the `/fixed/`

File: core/src/main/java/org/nervos/ckb/transaction/CellsWithLock.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (duanyytop) [reply core/src/main/java/org/nervos/ckb/transaction/CellsWithLock.java]: Updated. https://github.com/nervosnetwork/ckb-sdk-java/pull/202/commits/ad9880df962ec2fa0f1461c2663fb69fcae59cab

File: serialization/src/main/java/org/nervos/ckb/type/fixed/Option.java (hunk 1 above)
Comment (duanyytop) [reply serialization/src/main/java/org/nervos/ckb/type/fixed/Option.java]: Updated. 
* https://github.com/nervosnetwork/ckb-sdk-java/pull/202/commits/031038affd37b18242a6fe0339dfc9736bc3c3f7

* https://github.com/nervosnetwork/ckb-sdk-java/pull/202/commits/55ad71a8808a9daae619bebb16844617160655a1

**Pull request merged**